timeseries/src/autogluon/timeseries/models/gluonts/abstract_gluonts.py

```diff
@@ -247,7 +247,8 @@ def predict(self, data: TimeSeriesDataFrame, quantile_levels: List[float] = None
                         inplace=True,
                     )
 
-        return df
+        # Make sure the item_ids are sorted in the same order as in data
+        return df.loc[list(data.iter_items())]
```

Comment:
If this is the reason for the regression, didn't we also have predictions sorted incorrectly in v0.5.1? Which commit specifically led to the regression?

Reply:
The source of the problem is the new dataset splitter: we converted item IDs to strings (from ints), so the sorted order is `["1_[None:None]", "100_[None:None]", "101_[None:None]", ..., "999_[None:None]"]` instead of `[1, 2, 3, ..., 999]` before.

In any case, I think that preserving the item order during prediction is what the users would expect from the model anyway.

Reply:
all good. would you like to wait to rebase this on #2093 ? just so you can check offline `df.loc[data.item_ids]` doesn't break?

Reply:
Sure, will do that once you merge #2093 into master.